### DISCUSSION TITLE
Frequent Crashes

### DISCUSSION THREAD

**Discussion by michaelgrosser** [mbround18/valheim-docker#221] [Help!]
I'm running the server with Valheim Plus on a t3.medium on AWS (2 CPU, 4GB RAM), and I'm seeing a lot of crashes. The error I see the most is this one: 

```
valheim_1  | exception.c:79: Could not allocate 136 bytes assembly:<unknown assembly> type:<unknown type> member:(null) signature:<none>
valheim_1  |
valheim_1  | Caught fatal signal - signo:6 code:-6 errno:0 addr:0x3e80000095c
valheim_1  | threadpool.c:354: Could not allocate 58 bytes assembly:<unknown assembly> type:<unknown type> member:(null) signature:<none>
valheim_1  |
valheim_1  | threadpool.c:354: Could not allocate 58 bytes assembly:<unknown assembly> type:<unknown type> member:(null) signature:<none>
valheim_1  |
valheim_1  | Caught fatal signal - signo:6 code:-6 errno:0 addr:0x3e80000095c
valheim_1  | Caught fatal signal - signo:6 code:-6 errno:0 addr:0x3e80000095c
```

But there are also occasional out of memory crashes despite the server still having plenty of free memory. Anyone else seeing behavior like this? I'm not sure if the Valheim Linux server client is just buggy or if there is something else afoot.

**Comment by mbround18** [comment]
Are you out of disk space? 0.o

**Reply by michaelgrosser** [reply]
No, there's still about 10GB free.

**Reply by CosmicHorrorDev** [reply]
You wouldn't happen to be running a 32-bit OS, would you? The memory limit for a single process in a 32-bit system varies but can end up being a bit over 2GB sometimes

**Reply by michaelgrosser** [reply]
No, it's running Amazon Linux 2, which is 64 bit. 

Just to sanity check, the output of `uname -m` is `x86_64` as well.

**Reply by CosmicHorrorDev** [reply]
I've really got no clue. All signs seem to point to valheim at least thinking there's no RAM since it got a SIGABT. I've never personally seen those error messages, but my dedicated server has plenty of memory to use.

Those allocation sizes seem small, so if there really is plenty of free memory then I'd be surprised that some sort of heap overflow occurred.

**Reply by mbround18** [reply]
@michaelgrosser I am finally home and can test this, do you have a terraform, cloudformation, or similar environment script I could utilize? Or can you share your configurations (with screenshots and appropriate censoring)  of your EC2 instance?

**Reply by michaelgrosser** [reply]
@mbround18 I'd just created this instance quick in the console. It's a t3.medium created with the current AWS Linux 2 ECS AMI, which is `ami-0ecd34837cf9fa094`. I had it configured for Unlimited CPU credits and 50GB of disk. Beyond that, nothing special, only open ports were 22 for ssh and the required Valheim ports. 

On a separate note though, I moved up to a t3.large and have not had further issues. So, it seems like it was probably a resource issue despite the server still showing free RAM at the time of crashes.

**Reply by CosmicHorrorDev** [reply]
Hmm, @mbround18 you think we should bump up the memory warning limit? I wonder it that's also part of @srepollock 's problem

**Reply by mbround18** [reply]
it should only take about 2gb of memory to run with ~1-5 players than 3-4 gb for 5-10 players. I know some others were having issues on AWS as well, I wanted to ask are you using shared resources, spot instance, or have any feature turned on that might limit your use of ram? Im going through the console now and there are a lot of features to reduce cost of running a `t2.medium` but would shear off a bit of memory when another system on their services need it.

**Reply by michaelgrosser** [reply]
Nope. Just a vanilla On Demand t3.medium. With the t family of instances the only thing that should be burstable is CPU, however I turned on "Unlimited" CPU credits, and I never even started bursting into CPU credits anyway. RAM should be fixed quantity. A spot instance would just take your instance away if the spot market outbid you, but also shouldn't have any difference in available resources while you're running it.

**Comment by srepollock** [comment]
I was running this on a t2.medium for clarification, 2 cores, 4 gb ram and
15GB storage; requirements weren’t an issue, I could not connect from my PC
instance of Valheim to the ec2 Docker server
